docs/spec.md

```diff
@@ -704,195 +708,375 @@ expressions must not be preceded by whitespace before the `[` character,
-   the expressions of an expression list, including the values of map
-   literal definitions.
-
-4. `WS` is allowed within the expression of an expression list if the
-   expression is surrounded by `()`, `[]` or `{}`, `e.g. [ ( 2 + 3 ) ]`,
-   not `WS` directly after `[` and within `( … )`
-
-5. `WS` can be freely used in single expressions for assignments,
-   inferred declarations, return statements, `if` conditions and `while`
-   conditions, as well as within parentheses `(…)`
-
-Examples:
-
-    print -5      // -5
-    print - 5     // invalid
-    print 2-1     // 1
-    print 2 -1    // 2 -1
-    print 2 - 1   // invalid
-    a := 2 - 1    // valid!
-
-    arr := ["a" "b"]
-    print arr[1]        // b
-    print arr [1]       // [a b] [1]
-    arr [0] = "A"       // invalid
-    arr2 :=[ 1   ]      // valid
-    arr3 := [[1][2]]    // valid
-    arr3 := [[1] [ 2] ] // valid
-    arr3 := [1 + 1 ]    // invalid
-
-    m1 := { age:3+6 name:"mary"+"anne" }     // valid
-    m2 := {age:  12 name:"mary"}             // valid
-    m1.address = "10 Downing" + "Street"     // valid
-    m3 := {address: "10 Downing" + "Street"} // invalid
-
-    func add:num n1:num n2:num
-        return n1 + n2 // valid
-    end
-    print (add 1 2)  // 3
-    print add 1 2    // invalid
+Whitespace in Evy is used to separate different parts of a program.
+There are two types of whitespace in Evy: vertical whitespace and
+horizontal whitespace.
+
+### Vertical whitespace
+
+Vertical whitespace is a sequence of one or more newline characters that
+can optionally contain comments. It is used to terminate or end basic
+statements in Evy. A basic statement is a statement that cannot be
+broken up into smaller statements, such as a variable declaration,
+an assignment or a function call.
+
+Evy does not allow multiple statements on a single line. For example,
+the following code is invalid because it contains two statements, a
+declaration and a function call, on one line:
+
+`x := 1      print x`
+
+It is also not possible to break up a single basic statement over more
+than one line. For example, the following code is invalid because the
+arithmetic expression `1 + 2` is split over two lines:
+
+```
+x := 1 +
+     2
+```
+
+The rule that basic statements cannot be split across multiple lines has
+one exception: Array literals and map literals can be broken up over
+multiple lines, as long as each line is a complete expression. For
+example, the following code is valid because it is a declaration with a
+multiline map literal:
+
+```evy
+person := {
+    name:"Jane Goddall"
+    born:1934
+}
+print person
+```
+
+## Horizontal whitespace
+
+_Horizontal whitespace_ is a sequence of tabs or spaces that is used to
+separate elements in lists. Lists include the argument list to a
+function call, the element list of an array literal, and the value in
+the key-value pairs of a map literal. However, horizontal whitespace
+is not allowed _within_ this list elements.
```

Comment:
This doesn't mention horizontal whitespace being valid in an expression of an assignment or inferred decl:

    a := 1 + 2

Reply:
ok, good point. I've added a paragraph below, please let me know what you think.

Reply:
Yeah, that's good